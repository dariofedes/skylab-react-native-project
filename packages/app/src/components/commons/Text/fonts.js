export default {
    bold: 'RobotoMono-Bold',
    italic: 'RobotoMono-Italic'
}