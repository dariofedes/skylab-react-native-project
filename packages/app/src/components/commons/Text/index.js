import React from 'react'
import { Text } from 'react-native'
import PropTypes from 'prop-types';

import TextStyles from './styles'

const $Text = ({ children, onPress, style, type }) => {
    return (
        <Text
            onPress={onPress}
            style={{ ...TextStyles[type], ...style }}
        >{children}</Text>
    )
}

$Text.propTypes = {
    onPress: PropTypes.func,
    style: PropTypes.object,
    type: PropTypes.oneOf(['default', 'title', 'subtitle', 'link'])
}

$Text.defaultProps = {
    onPress: null,
    style: {},
    type: 'default'
}

export default $Text