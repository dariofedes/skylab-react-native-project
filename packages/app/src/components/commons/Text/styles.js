import { StyleSheet } from "react-native";

let styles = StyleSheet.create({
    default: {
        fontWeight: '300',
        fontSize: 18,
        color: '#333',
    },
    title: {},
    subtitle: {},
})

styles.title = {
    ...styles.default,
    margin: 30,
    alignSelf: 'center',
    fontSize: 30,
    fontWeight: '700'
}

styles.subtitle = {
    ...styles.default,
    margin: 10,
    fontSize: 24,
    fontWeight: '500'
}

export default styles