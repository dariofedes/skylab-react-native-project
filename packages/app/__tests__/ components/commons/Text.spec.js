import React from 'react'
import Renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'

import Text from '@skylab/app/src/components/commons/Text'

describe('Text component', () => {
    describe('Validate props', () => {
        it('should render correctly without children', () => {
            const renderer = Renderer.create(<Text />)
            const output = renderer.root
    
            expect(output.props.onPress).toStrictEqual(null)
            expect(output.props.style).toStrictEqual({})
            expect(output.props.type).toStrictEqual('default')
        })
    
    
        it('Should render correctly with default props', () => {
            const renderer = Renderer.create(<Text>Hello</Text>)
            const output = renderer.root
    
            expect(output.props.onPress).toStrictEqual(null)
            expect(output.props.style).toStrictEqual({})
            expect(output.props.type).toStrictEqual('default')
        })
    
        it('Should render correctly with some props (mock function)', () => {
            const onPressMock = jest.fn()
            const renderer = Renderer.create(
                <Text
                    onPress={onPressMock}
                    style={{ color: 'pink' }}
                    type='subtitle'
                >Hello</Text>
            )
            const output = renderer.root
    
            expect(output.props.onPress).toStrictEqual(onPressMock)
            expect(output.props.style).toStrictEqual({ color: 'pink' })
            expect(output.props.type).toStrictEqual('subtitle')
        })
    
        it('Should render correctly with some props (expect any function)', () => {
            const renderer = Renderer.create(
                <Text
                    onPress={() => true}
                    style={{ color: 'pink' }}
                    type='subtitle'
                >Hello</Text>
            )
            const output = renderer.root
    
            expect(output.props.onPress).toStrictEqual(expect.any(Function))
            expect(output.props.style).toStrictEqual({ color: 'pink' })
            expect(output.props.type).toStrictEqual('subtitle')
        })
    })

    describe('Snapshots', () => {
        it('should render correctly without props', () => {
            const renderer = new ShallowRenderer();
            const output = renderer.render(<Text>Hello</Text>);

            expect(output).toMatchSnapshot();
        })

        it('should render correctly with props', () => {
            const renderer = new ShallowRenderer();
            const output = renderer.render(
                <Text
                    onPress={() => true}
                    style={{ color: 'pink' }}
                    type='subtitle'
                >Hello</Text>
            );
            
            expect(output).toMatchSnapshot()
        })
    })
})