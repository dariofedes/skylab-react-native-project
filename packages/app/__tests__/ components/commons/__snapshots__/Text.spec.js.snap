// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Snapshots should render correctly with props 1`] = `
<Text
  onPress={[Function]}
  style={
    Object {
      "color": "pink",
      "fontFamily": "RobotoMono-Medium",
      "fontSize": 24,
      "fontWeight": "500",
      "margin": 10,
    }
  }
>
  Hello
</Text>
`;

exports[`Text component Snapshots should render correctly without props 1`] = `
<Text
  onPress={null}
  style={
    Object {
      "color": "#333",
      "fontFamily": "RobotoMono-Medium",
      "fontSize": 18,
      "fontWeight": "300",
    }
  }
>
  Hello
</Text>
`;
